src/goto-cc/compile.cpp

```
@@ -634,29 +635,32 @@ bool compilet::write_bin_object_file(
 /// \return true on error, false otherwise
 bool compilet::parse_source(const std::string &file_name)
 {
-  if(parse(file_name))
+  language_filest language_files;
+  language_files.set_message_handler(get_message_handler());
+
+  if(parse(file_name, language_files))
     return true;
 
-  if(typecheck()) // we just want to typecheck this one file here
+  // we just typecheck one file here
+  if(language_files.typecheck(symbol_table))
+  {
+    error() << "CONVERSION ERROR" << eom;
     return true;
+  }
 
-  if((has_suffix(file_name, ".class") ||
-      has_suffix(file_name, ".jar")) &&
-     final())
```

This got silently lost.

---

Ok, put into separate commit